### PR TITLE
build: update Go and K8s api version

### LINKED ISSUE #309: Update go to 1.22 and kubernetes api to latest version
Let's move to go 1.22 and upgrade Kubernetes API to the latest version, in go mod and CI also.

### PR DESCRIPTION
**Pull request by ShravaniVangur**
The commit updates the version of Go from 1.21 to 1.22.0  and updates the Kubernetes api versions accordingly. 

Resolves #309

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -11,9 +11,9 @@ require (
 	github.com/spf13/cobra v1.8.1
 	github.com/stretchr/testify v1.9.0
 	gopkg.in/yaml.v3 v3.0.1
-	k8s.io/api v0.29.3
-	k8s.io/apimachinery v0.29.3
-	k8s.io/client-go v0.29.3
+	k8s.io/api v0.30.2
+	k8s.io/apimachinery v0.30.2
+	k8s.io/client-go v0.30.2
```
Comment (Madhu-1) [inline go.mod:16]: This also needs to be changed in go.sum as well, i don't see that, did you run `go mod tidy` after updating these values?

File: .github/workflows/set-up-go/action.yaml [hunk 2]
```diff
@@ -7,4 +7,4 @@ runs:
     - name: setup golang
       uses: actions/setup-go@v5
       with:
-        go-version: "1.21"
+        go-version: "1.22.0"
```
Comment (Madhu-1) [inline .github/workflows/set-up-go/action.yaml:10]: @subhamkrai how about updating this to use go.mod so that we don't need to change in multiple places?

**Review by subhamkrai — CHANGES_REQUESTED** [review]
@ShravaniVangur  you need to sign-off your changes you can read here https://rook.github.io/docs/rook/latest/Contributing/development-flow/#commit-structure

**Comment by subhamkrai** [comment]
> @ShravaniVangur you need to sign-off your changes you can read here https://rook.github.io/docs/rook/latest/Contributing/development-flow/#commit-structure



> @ShravaniVangur you need to sign-off your changes you can read here https://rook.github.io/docs/rook/latest/Contributing/development-flow/#commit-structure

@ShravaniVangur can you please fix the CI error thanks

**Review by subhamkrai — APPROVED** [review]

**Pull request merged**